serif_swing/src/main/kotlin/serif_swing/App.kt

```diff
@@ -160,7 +191,21 @@ class SwingChatRoom(val transition: (MatrixState, Boolean) -> Unit, val panel: J
         val onSend: (ActionEvent) -> Unit = {
             val text = message_field.text
             message_field.text = ""
-            transition(m.sendMessage(text), true)
+            val res =
+            if(replied_event_id == "" && edited_event_id == "") {
```

Comment:
I think you could use a when expression to pattern match on the cases, though I'm not positive.

Reply:
You can indeed.